packages/ui/core-components/src/lib/atoms/inputs/dropdown/Dropdown.svelte

```diff
@@ -167,23 +175,26 @@
 	let searchIdx = 0;
 	let finalQuery;
 
-	const updateQuery = debounce(async () => {
+	const _updateQuery = () => {
 		searchIdx++;
 		if (search && hasQuery) {
 			const targetIdx = searchIdx;
 			const searchQuery = query.search(search, 'label');
 			if (searchQuery.hash !== finalQuery?.hash) {
-				await searchQuery.fetch();
-				if (targetIdx === searchIdx) {
-					finalQuery = searchQuery;
-					forceSort();
-				}
+				resolveMaybePromise(() => {
+					if (targetIdx === searchIdx) {
+						finalQuery = searchQuery;
+						forceSort();
+					}
+				}, searchQuery.fetch());
 				// await tick();
 			}
 		} else {
 			finalQuery = query ?? data;
 		}
-	}, 100);
+	};
+
+	const updateQuery = browser ? debounce(_updateQuery, 100) : _updateQuery;
```

Comment:
We need to make a browserDebounce function in the SDK lib that can be used in these cases - too much repetition of the same pattern for the same reason